packages/devtools_app/lib/src/inspector/flutter/story_of_your_layout/flex.dart

```diff
@@ -319,23 +373,54 @@ class _StoryOfYourFlexWidgetState extends State<StoryOfYourFlexWidget>
     _onInspectorSelectionChanged();
   }
 
-  void _updateProperties() async {
-    if (_previousProperties != null) {
-      entranceController.reverse();
-    } else {
-      entranceController.forward();
-    }
-  }
-
-  Future<void> _onTap(LayoutProperties properties) async {
-    highlighted = properties;
-    final node = properties.node;
+  Future<void> _onTap(LayoutProperties selected) async {
+    highlighted = selected;
+    final node = selected.node;
     inspectorController.refreshSelection(node, node, false);
   }
 
-  Future<void> refresh() async {
-    setProperties(await fetchFlexLayoutProperties());
-    setState(() {});
+  Future<void> updateChildFlex(
+      LayoutProperties oldProperties, LayoutProperties newProperties) async {
+    final index = properties.children.indexOf(oldProperties);
+    if (index != -1) {
+      print('updating child properties $index');
+      final newChildren = properties.children.toList();
+      newChildren[index] = newProperties;
+      // Recompute the sizes of the children.
+      double mainAxisSizes = 0.0;
+      num flexes = 0;
+      for (var child in newChildren) {
+        if (!child.hasFlexFactor) {
+          mainAxisSizes += child.dimension(properties.direction);
+        } else {
+          flexes += child.flexFactor;
+        }
+      }
+      final remainingSpace = properties.mainAxisDimension - mainAxisSizes;
+      // Lay out flex space.
+      for (var i = 0; i < newChildren.length; i++) {
+        final child = newChildren[i];
+        if (child.hasFlexFactor) {
+          final flexFraction = remainingSpace * child.flexFactor / flexes;
+          final width = properties.isMainAxisHorizontal
+              ? flexFraction
+              : child.dimension(Axis.horizontal);
+          final height = properties.isMainAxisVertical
+              ? flexFraction
+              : child.dimension(Axis.vertical);
+          newChildren[i] = child.copyWith(
+            size: Size(width, height),
+          );
+        }
+      }
+      newProperties = properties.copyWith(children: newChildren);
+      _changeProperties(newProperties);
+    }
```

Comment:
this is nice Dave.
One other option instead of reimplementing the flex layouting process here is we can actually just do the fetching first and animate between those two values.
So this function will not need the newProperties parameter,
Instead:
```dart
Future<void> updateChildFlex(int idx) async {
  final oldProperties = properties.children[idx];
  final newProperties = await fetchFlexLayoutProperties();
  _changeProperties(newProperties);
  setState(() => _properties = newProperties);
}
```

This way, we don't have to maintain the flex computing process and the properties here is always going to be sync with the node in the Flutter app.

Reply:
the down side for this approach that I mentioned would be: we have to wait until the fetching is done from the framework side for the animation to start.
What do you think? @jacob314

Reply:
There are sometimes several seconds of latency that I'm observing.  In some cases, it becomes unusable.